expo/scheduler/views.py

```diff
@@ -906,6 +782,143 @@ def add_event(request, eventitem_id, event_type='Class'):
+                                                urlconf='scheduler.urls',
+                                                args=[event_type,
+                                                      scheduler_event_id]))
+        else:
+            template = 'scheduler/event_schedule.tmpl'
+            return render(request, template, {
+                'eventitem': get_event_display_info(
+                    item.eventitem.eventitem_id),
+                'form': event_form,
+                'event_type': item.event_type_name})
+
+    else:
+        return edit_event_display(request, item)
+
+
+def edit_event_display(request, item, errorcontext=None):
+    from gbe.models import Performer
+
+    template = 'scheduler/event_schedule.tmpl'
+    context = {'user_id': request.user.id,
+               'event_id': item.id,
+               'event_edit_url': reverse('edit_event',
+                                         urlconf='scheduler.urls',
+                                         args=[item.event_type_name,
+                                               item.id])}
+    context['eventitem'] = get_event_display_info(item.eventitem.eventitem_id)
+
+    initial = {}
+    initial['duration'] = item.duration
+    initial['day'] = item.starttime.strftime("%Y-%m-%d")
+    initial['time'] = item.starttime.strftime("%H:%M:%S")
+    initial['description'] = item.as_subtype.sched_payload['description']
+    initial['title'] = item.as_subtype.sched_payload['title']
+    initial['location'] = item.location
+
+    allocs = ResourceAllocation.objects.filter(event=item)
+    workers = [Worker.objects.get(id=a.resource.id)
+               for a in allocs if type(a.resource.item) == WorkerItem]
+    teachers = [worker for worker in workers if worker.role == 'Teacher']
+    moderators = [worker for worker in workers if worker.role == 'Moderator']
+    panelists = Performer.objects.filter(worker__role='Panelist',
+                                         worker__allocations__event=item)
+    staff_leads = [worker for worker in workers if worker.role == 'Staff Lead']
+
+    # Set initial values for specialized event roles
+    if len(teachers) > 0:
+        initial['teacher'] = teachers[0].item
+
+    elif item.event_type_name == 'Class':
+        try:
+            initial['teacher'] = item.as_subtype.teacher
+        except:
+            pass
+
+    if len(moderators) > 0:
+        initial['moderator'] = moderators[0].item
+    if len(panelists) > 0:
+        initial['panelists'] = panelists
+    if len(staff_leads) > 0:
+        initial['staff_lead'] = staff_leads[0].item
+
+    context['event_type'] = item.event_type_name
+
+    if validate_perms(request, ('Volunteer Coordinator',), require=False):
+        if (item.event_type_name == 'GenericEvent' and
+            item.as_subtype.type == 'Volunteer'):
+
+            context.update(get_worker_allocation_forms(item, errorcontext))
+            context.update(show_potential_workers(item))
```

Comment:
Causes an error when I try to edit a volunteer opportunity that has been scheduled:
http://localhost:8282/scheduler/edit/GenericEvent/<id of volunteer opportunity>